rs/replicated_state/src/canister_state/execution_state.rs

```diff
@@ -446,6 +451,7 @@ pub struct ExecutionState {
     ///
     /// The latter property ensures that compilation for queries is cached
     /// properly when loading a state from checkpoint.
+    #[validate_eq(CompareWithValidateEq)]
     pub wasm_binary: Arc<WasmBinary>,
 
     /// The persistent heap of the module. The size of this memory is expected
```

Comment:
The two `Memory` just below should be recursive and ignore the `PageMap` inside. 

Furthermore, I am not sure how big `exported_globals`, `exports` and `metadata` are, but they sound like they could be large.